net/WebSocketHandler.hpp

```diff
@@ -252,8 +252,9 @@ class WebSocketHandler : public ProtocolHandlerInterface
             if (!_shuttingDown)
                 sendCloseFrame(statusCode, statusMessage);
             socket->closeConnection();
-            socket->getInBuffer().clear();
             socket->ignoreInput();
+            assert(socket->getInBuffer().empty() &&
```

Comment:
Not convinced by this. A websocket can have lots of queued input, and yet we need to shut it down because one of the top messages is malformed eg. Would prefer the clear after the assert if we must have it - but it will cause grief I think.

Reply:
`ignoreInput()` clears the buffer. Since I removed the redundant `socket->getInBuffer().clear();` (2 lines above), I put this assert to signal that removing the clear is safe and to defend against regressions.

Unless I missed your point.

Reply:
Ah - I missed that; this is fine :-) thanks.